backend/src/main/java/ch/puzzle/pcts/service/validation/CertificateValidationService.java

```diff
@@ -5,91 +5,43 @@
 import ch.puzzle.pcts.model.certificate.CertificateType;
 import ch.puzzle.pcts.model.error.ErrorKey;
 import ch.puzzle.pcts.service.persistence.CertificatePersistenceService;
-import java.math.BigDecimal;
+import ch.puzzle.pcts.service.validation.util.UniqueNameValidationUtil;
 import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 
 @Service
-public class CertificateValidationService {
+public class CertificateValidationService extends ValidationBase<Certificate> {
+
     private final CertificatePersistenceService persistenceService;
 
     public CertificateValidationService(CertificatePersistenceService persistenceService) {
         this.persistenceService = persistenceService;
     }
 
-    public void validateOnCreate(Certificate certificate) {
-        validateIfIdIsNull(certificate.getId());
-        validateName(certificate.getName());
-        validatePoints(certificate.getPoints());
-        validateCertificateType(certificate.getCertificateType());
-    }
-
-    public void validateOnGetById(Long id) {
-        validateIfExists(id);
-    }
-
+    @Override
     public void validateOnUpdate(Long id, Certificate certificate) {
-        validateIfExists(id);
-        validateIfIdIsNull(certificate.getId());
-        validateName(certificate.getName());
-        validatePoints(certificate.getPoints());
+        super.validateOnUpdate(id, certificate);
         validateCertificateType(certificate.getCertificateType());
-    }
-
-    public void validateOnDelete(Long id) {
-        validateIfExists(id);
-    }
-
-    public void validateCertificateType(CertificateType certificateType) {
-        if (certificateType == null) {
-            throw new PCTSException(HttpStatus.BAD_REQUEST,
-                                    "Certificate type must not be null.",
-                                    ErrorKey.CERTIFICATE_TYPE_IS_NULL);
-        }
-        if (certificateType != CertificateType.CERTIFICATE) {
-            throw new PCTSException(HttpStatus.BAD_REQUEST,
-                                    "Certificate type is not certificate.",
-                                    ErrorKey.CERTIFICATE_TYPE_IS_NOT_CERTIFICATE);
+        if (UniqueNameValidationUtil
+                .nameExcludingSelfAlreadyUsed(id, certificate.getName(), persistenceService::getByName)) {
```

Comment:
[MED–TICK]: create a ticket for this issue: add warn logs for validation failures